### PR TITLE
fix: Fix/js object ads issues

### PR DESCRIPTION
**Pull request by Irongade**
This PR addresses issues raised when migrating JsObjects to new design system.

Issue link here: https://www.notion.so/appsmith/f353d8c6bd664f79ad858a42010cdfc8?v=f04cde23f6424aeb9d5a6e389cd172bd&p=6ff9af061f354a569b160157453bfb3f&pm=s

- Bug fix (non-breaking change which fixes an issue)

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx [hunk 1]
```diff
@@ -133,9 +133,45 @@ const italicInfoStyles = {
   fontStyle: "italic",
 };
 
-const StyledTooltip = styled(Tooltip)`
-  & .hide-me {
-    display: none;
+const StyledTooltip = styled.span<{ width?: number }>`
```
Comment (rohitagarwal88) [inline app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx:136]: @Irongade : why have you created your own tooltip?

Comment (Irongade) [reply app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx:136]: The Tooltip afforded by the design system, did not fulfill the unique positional requirements of that feature. Hence, I created a custom one based on the Tooltip from the design system. I believe we should move forward with this solution for now.

Comment (rohitagarwal88) [reply app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx:136]: @Irongade : can you explain in detail the unique positional requirements?

File: app/client/src/components/editorComponents/Debugger/ErrorLogs/components/LogEntityLink.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ export default function LogEntityLink(props: LogItemProps) {
       }
       // If the source is a JS action.
       else if (props.source.type === ENTITY_TYPE.JSACTION) {
-        return JsFileIconV2(16, 16);
```
Comment (tanvibhakta) [inline app/client/src/components/editorComponents/Debugger/ErrorLogs/components/LogEntityLink.tsx:54]: What is this for?

File: app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx (hunk 1 above)
Comment (tanvibhakta) [reply app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx:136]: > can you explain in detail the unique positional requirements?
seconded. I'd love to understand what the limitations of the tooltip from the design-system is so we can improve it. Also, have you taken a look at the options offered by [rc-tooltip](https://www.npmjs.com/package/rc-tooltip), which is what our tooltip is based on, and see if the documentation there might help you? 

@Irongade even if there are positional requirements that are unique here, you must inherit from the design-system tooltip and only then apply your positioning fixes.

File: app/client/src/components/editorComponents/form/fields/TextField.tsx [hunk 3]
```diff
@@ -19,6 +19,7 @@ class TextField extends React.Component<
         {...this.props}
         isDisabled={this.props.isDisabled}
         noValidate
+        size={36}
```
Comment (tanvibhakta) [inline app/client/src/components/editorComponents/form/fields/TextField.tsx:22]: Why is a size being set for this? This component is being used in multiple places. Have you validated that this size works for all the usages?

File: app/client/src/components/utils/FlagBadge.tsx [hunk 4]
```diff
@@ -1,18 +1,8 @@
...
-  text-transform: uppercase;
-  font-size: 10px;
-  font-weight: 600;
-  margin-right: 10px;
-`;
+import { Tag } from "design-system";
 
 function FlagBadge(props: { name: string }) {
-  return <Flag>{props.name}</Flag>;
+  return <Tag isClosable={false}>{props.name}</Tag>;
```
Comment (tanvibhakta) [inline app/client/src/components/utils/FlagBadge.tsx:5]: This component now seems like it's a simple wrapper. Can we replaces all usages of FlagBadge with Tag instead?

**Review by tanvibhakta — COMMENTED** [review]
Apart from the above comments, 

1. There's a border when you hover the text area here. This must be removed, but without affecting the hover for the regular text inputs. 
Screenshot 2023-05-10 at 11 03 38 PM

2. The border radius for the table here 
Screenshot 2023-05-10 at 11 05 51 PM

3. Spacing for the Code | Settings tabs need to match the spacing in the API pane 

4. On release, when the function is invalid, the select and run button is disabled, but on our branch the select is enabled and only the run button is disabled. Okay honestly, even though that select looks disabled it's still usable, so I'd like @vasanthappsmith 's clearance on this. 

5. Let's make these arrow buttons on the debugger size small. 
Screenshot 2023-05-10 at 11 16 47 PM


I've added all of these points to the notion document as well so that it's easier to have a discussion about them.

**Comment by tanvibhakta** [comment]
Screenshot 2023-05-10 at 11 25 47 PM

Removing of the logo background and border should not have been done because it doesn't match the entity explorer + the contrast doesn't work.

File: app/client/src/components/editorComponents/Debugger/ErrorLogs/components/LogEntityLink.tsx (hunk 2 above)
Comment (Irongade) [reply app/client/src/components/editorComponents/Debugger/ErrorLogs/components/LogEntityLink.tsx:54]: Thats for displaying the Js File Icon in the debugger.

File: app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx (hunk 1 above)
Comment (Irongade) [reply app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx:136]: Explaining this via chat will be challenging, I would rather show both of you. Please let's connect later today to discuss this further.

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 5]
```diff
@@ -267,9 +267,9 @@ export const CodemirrorHintStyles = createGlobalStyle<{
     padding-left: 20px;
   }
   .CodeMirror-lint-mark-warning{
-    background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAFlSURBVHgBrVLRTcMwELWDg8QHtJ2AfFZtJNIJgAnoCOkEZIMkIzBBwwSwAR0hSG3VzzABhahCSmKbOyVxLJQEJPp+7DvfvXe+O0L+CdrmTG37RkjpwOMV2pLSt1PGorM4TnoJpOMM06JYEinnpB2RaZqhTqQIvhzHyvP8Ba4W6UfCTXM2iuM9GkbthWT/D8kI6yTLlrVh1OpwuLVTEhLDv0NlS/mAPkVB6Rz7pAiyonB1CXCuBut1gCSQHA62W8+g9FmPwSbjySoJS38ENS+dTPbnQII2qPlAFOgx9YQM0gGhNVh0jLupgNIEqmgqKMsOUJkIQbCSD9seUinvVQzshiLAP0O639RH79LpdABEHtwJfAdZr6VeYVFE1VdKfNr2U88C/UR0sdksKvESnLEFKCS/ZWIM51yNWBHgZgnObyHgsTMbRikOh9lot0uUqy3ufTy2DMZcaNplpfoKexBDM1fk2PgGIxqcuvBfxngAAAAASUVORK5CYII=");
```
Comment (Irongade) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:270]: Why was this added? @tanvibhakta 

Reference: https://www.notion.so/appsmith/JS-Objects-6ff9af061f354a569b160157453bfb3f?pvs=4#d43e91d66ebd4e0cbe4eb9644100a379

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 6]
```diff
@@ -267,9 +267,9 @@ export const CodemirrorHintStyles = createGlobalStyle<{
...
-  }
-  .appsmith-light-theme .CodeMirror-lint-message-error {
```
Comment (Irongade) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:272]: Why was this added? @AmanAgarwal041 

Also please note that there is no difference between the image here and the one above.

Reference: https://www.notion.so/appsmith/JS-Objects-6ff9af061f354a569b160157453bfb3f?pvs=4#d43e91d66ebd4e0cbe4eb9644100a379
